EventAggregator/EventAggregator.cs

```diff
@@ -424,20 +544,109 @@ public bool HandlesMessage(object message)
                 return handled;
             }
 
-            public void TryHandle<TListener>(object target, object message, out bool wasHandled)
+            public bool TryHandle<TListener>(object target, object message)
                 where TListener : class
             {
-                wasHandled = false;
                 if (target == null)
                 {
-                    return;
+                    return false;
                 }
 
-                if (!Handles<TListener>() && !HandlesMessage(message)) return;
-
+                if (!Handles<TListener>() && !HandlesMessage(message)) return false;
+#if !ASYNC && !UNWRAP_EX
                 _handlerMethod.Invoke(target, new[] {message});
-                wasHandled = true;
+#elif ASYNC && !UNWRAP_EX
+                if (TypeHelper.IsAssignableToGenericType(target.GetType(), typeof(IListenerAsync<>)))
+                {
+                    ((dynamic) _handlerMethod.Invoke(target, new[] {message})).Wait();
+                }
+                else
+                {
+                    _handlerMethod.Invoke(target, new[] {message});
+                }
+#elif ASYNC
+                if (TypeHelper.IsAssignableToGenericType(target.GetType(), typeof(IListenerAsync<>)))
+                {
+                    try
+                    {
+                        ((dynamic) _handlerMethod.Invoke(target, new[] {message})).Wait();
+                    }
+                    catch (AggregateException ex)
+                    {
+#if NETFX_CORE || WINDOWS_PHONE || NET_CLIENT
+                        throw ex.InnerException;

```

Comment:
I added this method only for NET_CLIENT as WINDOWS_PHONE and NETFX_CORE platforms Exception do not have a InternalPreserveStackTrace method.